.editorconfig

```diff
@@ -0,0 +1,170 @@
+root = true
```

Comment:
[fix] We had a chat at some point, a while back (prob before your time), about doing away with the `ij_` wall-of-text for new projects (e.g. [BAL](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/blob/main/.editorconfig)). Could we do the same here please?  Especially as we can use PEP8 here.

Reply:
This issue was raised when creating the OpenAssetIO-Test-CMake project.  The `.clang-tidy` and `.editorconfig` from that is a boiled-down version of the ones in the OpenAssetIO repo, and may be a better reference.

Reply:
Replaced the `.editorconfig` and `.clang-tidy` files wholesale with the ones from the `Test-CMake` project.
[250560e](https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/3/commits/250560ee82c6a74db5db730bf8fc76df71be6c00)